src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -452,15 +452,12 @@ public void execute(LifecycleTransaction txn) throws IOException
     public AllSSTableOpStatus performCleanup(final ColumnFamilyStore cfStore, int jobs) throws InterruptedException, ExecutionException
     {
         assert !cfStore.isIndex();
-
-        if (nodeHasPendingRangesForKeyspace(cfStore))
+        Keyspace keyspace = cfStore.keyspace;
+        if (!StorageService.instance.getTokenMetadata().getPendingRanges(keyspace.getName(), FBUtilities.getBroadcastAddress()).isEmpty())
```

Comment:
I think the idea of wrapping that not-very pleasant to-read predicate in an auxiliary method had some value but I am not going to oppose to removing it strongly.